src/main/java/org/jvnet/hudson/update_center/MavenArtifact.java

```diff
@@ -136,14 +136,33 @@ public String getDigest() throws IOException {
         }
     }
 
+    /**
+     * Computes the SHA256 signature of the file.
+     */
+    public String getDigestSHA256() throws IOException {
+        try {
+            MessageDigest sig = MessageDigest.getInstance("SHA256");
+            FileInputStream fin = new FileInputStream(resolve());
+            byte[] buf = new byte[2048];
+            int len;
+            while ((len=fin.read(buf,0,buf.length))>=0)
```

Comment:
Not sure if worth refactoring, but this can create both signatures in a single pass through the data. So the method renamed to `get Digests` would return `Map<String, String>` with all the digests we produce.

Reply:
I like the idea of refactoring this, but went with the approach of passing in the sha type to use as a string and then calling it twice. Hope that's okay.